hal/network/lwip/esp32/esp32ncpnetif.cpp

```diff
@@ -124,9 +124,32 @@ err_t Esp32NcpNetif::initInterface() {
     return ERR_OK;
 }
 
+int Esp32NcpNetif::queryMacAddress() {
+    auto r = wifiMan_->ncpClient()->on();
+    if (r) {
+        LOG(TRACE, "Failed to initialize ESP32 NCP client: %d", r);
+        return r;
+    }
+    MacAddress mac = {};
+    if (!memcmp(interface()->hwaddr, mac.data, interface()->hwaddr_len)) {
+        // Query MAC address
+        r = wifiMan_->ncpClient()->getMacAddress(&mac);
+        if (r) {
+            LOG(TRACE, "Failed to query ESP32 MAC address: %d", r);
+            return r;
+        }
+        memcpy(interface()->hwaddr, mac.data, interface()->hwaddr_len);
+    }
+
+    return r;
+}
+
 void Esp32NcpNetif::loop(void* arg) {
     Esp32NcpNetif* self = static_cast<Esp32NcpNetif*>(arg);
     unsigned int timeout = 100;
+    auto r = self->queryMacAddress();
+    if (r) // Failed to query MAC Address
+        return;
```

Comment:
Error should be ignore here. Otherwise there is a possibility that due to some possibly intermittent communication error with ESP32 NCP, this whole thread will stop forever.

Reply:
Fixed.